pkg/chains/signing/x509/fsprovider.go

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package x509
+
+import (
+	"context"
+	"os"
+
+	"github.com/sigstore/cosign/pkg/providers"
+)
+
+func init() {
+	providers.Register("filesystem-custom-path", &filesystemCustomPath{})
+}
+
+type filesystemCustomPath struct{}
+
+var _ providers.Interface = (*filesystemCustomPath)(nil)
+
+var (
```

Comment:
Here's an explanation of what (I hope/think) I'm adding: 
I'm using a variable here and setting it to config `identity.token.file` value whenever that value is set in the `chains-config` configmap. In cosign, this is a hard-coded constant
(` FilesystemTokenPath`) in the `filesystem` implementation of provider.s.Interface. Here I'm registering an implementation of the cosign providers.Interface to allow for a dynamic value of the token path. This is not exposed to the users, they will either:
1) configure something like
    ```bash
      signers.x509.fulcio.provider: "filesystem"`
      signers.x509.identity.token.file: "/some/path"
     ```
     in this case the "filesystem-custom-path" will be set instead of cosign's "filesystem", 
     because this allows for injecting the non-hardcoded token path.
   
2) like this 
    ```bash
       # nothing set for fulcio.provider
       signers.x509.identity.token.file: "/some/path"
    ```
     in this case, the "filesystem-custom-path" will be set.
       
3) any other non-empty value for `signers.x509.fulcio.provider` will result in that 
        implementation being used (`google`, `spiffe`, `github`)

4) an empty value for both `signers.x509.fulcio.provider` and 
        `signers.x509.identity.token.file` results in all known providers being tried.

Reply:
It does make sense. It also seems like the least surprising behavior for those config options.